### ISSUE TITLE
BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding

### ISSUE THREAD

**Issue by ymgw55** [ymgw55/MLPapers#31]
# Information
### URL
https://arxiv.org/abs/1810.04805
### Authors
Jacob Devlin, Ming-Wei Chang, Kenton Lee, Kristina Toutanova
### Published at
Google
### Date of issue
11 Oct 2018

# Summary
BERTの論文. 近年の言語表現モデルとは異なり, BERTは全layer で左右の文脈を考えることで, ラベルなしの文章からbidirectional representation を pre-trainするように設計されている. 結果として, 1つの追加の出力層をpre-trainされたBERTモデルでfine-tuningするだけで,  タスクごとに十分にモデルのアーキテクチャを修正することなく, QAやlanguage inference といった, 幅広いタスクでSOTAモデルとなる. BERTはシンプルな発想でありながら実験から優れた能力を持つとわかる. 11のNLPタスクでSOTAを達成.
# Background
pre-trainedモデルを用いる際のよくある手法
* feature-based: pre-trained したパラメータを追加の特徴量とする; ELMo
* fine-tuning; GPT

いままでのモデルは一方向でpre-trained representations の能力が制限されていた. 

# Proposed Method
attention base のTransformer を用いることで双方向の文脈から学習できる.
pre-trainingとfine-tuningに分けて学習する. pre-trainingでは
* Masked LM: [MASK] or random or original のtokenを予測する学習.
* Next Sentence Prediction: [CLS] sentence A  [SEP] sentence B の形で入力し, A, B が連続かどうかを[CLS]の出力で予測.

の2つのタスクを行う. fine-tuningはタスクに応じて出力, 損失関数を工夫する必要がある. 
例えばQAでは回答箇所の始めと終わりのtoken のd 次元 embbeding をS, E, 出力をまとめたN x d行列を T として, 
N次元ベクトル softmax(TS) を各tokenが回答の始めとなる確率とする. 回答の終わりも類似した方法で決める. 
# Experiment
[GLUE bench mark](https://gluebenchmark.com/)で実験.
